pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -339,6 +353,10 @@ func hasContainerChanged(current, desired corev1.Container) (corev1.Container, b
 		updated = true
 	}
 
+	if hasSecurityContextChanged(current.SecurityContext, desired.SecurityContext) {
+		updated = true
```

Comment:
Why not returning right here? The whole container will be reset regardless of the results of the further comparisons.